lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -983,4 +1035,41 @@ namespace Js
         }
         return slotIndex;
     }
+
+    // static
+    Var SourceTextModuleRecord::ResolveOrRejectDynamicImportPromise(bool toResolve, Var value, ScriptContext *scriptContext, SourceTextModuleRecord *moduleRecord)
+    {
+        JavascriptPromiseCapability *promiseCapability = nullptr;
+        if (moduleRecord != nullptr)
+        {
+            promiseCapability = moduleRecord->GetPromiseCapability();
+        }
+
+        if (promiseCapability == nullptr)
+        {
+            promiseCapability = JavascriptPromise::NewPromiseCapability(scriptContext->GetLibrary()->GetPromiseConstructor(), scriptContext);
+        }
+
+        Var resolveOrRejectVar = toResolve ? promiseCapability->GetResolve() : promiseCapability->GetReject();
```

Comment:
+1, I also found this hard to read.

---
In reply to: [114906734](https://github.com/Microsoft/ChakraCore/pull/2913#discussion_r114906734) [](ancestors = 114906734)